sui/src/microbench_latency.rs

```diff
@@ -0,0 +1,558 @@
+        );
+
+        SingleTransactionKind::Call(MoveCall {
+            package: framework_obj_ref,
+            module: ident_str!("GAS").to_owned(),
+            function: ident_str!("transfer").to_owned(),
+            type_arguments: Vec::new(),
+            object_arguments: vec![object_ref],
+            shared_object_arguments: vec![],
+            pure_arguments: vec![bcs::to_bytes(&AccountAddress::from(recipient)).unwrap()],
+            gas_budget: 1000,
+        })
+    } else {
+        SingleTransactionKind::Transfer(Transfer {
+            recipient,
+            object_ref,
+        })
+    }
+}
+
+/// Creates an object for use in the microbench
+fn create_object(object_id: ObjectID, owner: SuiAddress, use_move: bool) -> Object {
+    if use_move {
+        Object::with_id_owner_gas_coin_object_for_testing(
+            object_id,
+            SequenceNumber::new(),
+            owner,
+            1,
+        )
+    } else {
+        Object::with_id_owner_for_testing(object_id, owner)
+    }
+}
+
+/// This builds, signs a cert and serializes it
+fn make_serialized_cert(
+    keys: &[(PublicKeyBytes, KeyPair)],
+    committee: &Committee,
+    tx: Transaction,
+) -> Vec<u8> {
+    // Make certificate
+    let mut certificate = CertifiedTransaction::new(tx);
+    for i in 0..committee.quorum_threshold() {
+        let (pubx, secx) = keys.get(i).unwrap();
+        let sig = AuthoritySignature::new(&certificate.transaction.data, secx);
+        certificate.signatures.push((*pubx, sig));
+    }
+
+    let serialized_certificate = serialize_cert(&certificate);
+    assert!(!serialized_certificate.is_empty());
+    serialized_certificate
+}
+
+fn make_authority_state(
+    store_path: &Path,
+    db_cpus: i32,
+    committee: &Committee,
+    pubx: &PublicKeyBytes,
+    secx: KeyPair,
+) -> (AuthorityState, Arc<AuthorityStore>) {
+    fs::create_dir(&store_path).unwrap();
+    info!("Open database on path: {:?}", store_path.as_os_str());
+
+    let mut opts = Options::default();
+    opts.increase_parallelism(db_cpus);
+    opts.set_write_buffer_size(256 * 1024 * 1024);
+    opts.enable_statistics();
+    opts.set_stats_dump_period_sec(5);
+    opts.set_enable_pipelined_write(true);
+
+    // NOTE: turn off the WAL, but is not guaranteed to
+    // recover from a crash. Keep turned off to max safety,
+    // but keep as an option if we periodically flush WAL
+    // manually.
+    // opts.set_manual_wal_flush(true);
+
+    let store = Arc::new(AuthorityStore::open(store_path, Some(opts)));
+    (
+        Runtime::new().unwrap().block_on(async {
+            AuthorityState::new(
+                committee.clone(),
+                *pubx,
+                Arc::pin(secx),
+                store.clone(),
+                genesis::clone_genesis_compiled_modules(),
+                &mut genesis::get_genesis_context(),
+            )
+            .await
+        }),
+        store,
+    )
+}
+
+fn make_gas_objects(
```

Comment:
Should we share some of these functions between benches? Otherwise we do 2x work on maintenance when code changes,

Reply:
100% agree.
It was just easier to iterate without worrying about parity. Will condense both in subsequent PR.